core/src/tools.rs

```diff
@@ -75,6 +46,12 @@ pub struct MightWriteContext<'a> {
     pub ir: &'a HarvestIR,
 }
 
+impl<'a> From<&'a HarvestIR> for MightWriteContext<'a> {
+    fn from(value: &'a HarvestIR) -> Self {
+        MightWriteContext { ir: value }
+    }
+}
```

Comment:
This `From` impl is going to become really inconvenient as soon as we add a second field to `MightWriteContext`. This should be a `new()` function instead.